messages/src/util.rs

```diff
@@ -15,3 +19,23 @@ pub fn bool_to_ascii(on: bool) -> u8 {
         b'0'
     }
 }
+
+pub fn read_bool(rest: &BitSlice<Msb0, u8>) -> Result<(&BitSlice<Msb0, u8>, bool), DekuError> {
```

Comment:
If Deku has a standard way of serialising `bool` then I think it's fine just to use that, rather than making extra effort to keep the current approach. I'm not too attached to how we currently do it.

Reply:
Ok. If you think this is a reasonable approach then I will clean this pr up into a mergeable state.

Reply:
I'd like to see how it looks for the firmware side, but so far it looks promising.